pulsar/message.go

```diff
@@ -120,19 +124,65 @@ type Message interface {
 type MessageID interface {
 	// Serialize the message id into a sequence of bytes that can be stored somewhere else
 	Serialize() []byte
+	// String the message id represented as a string
+	String() string
+	// Equals indicates to message IDs are equal
+	Equals(other MessageID) bool
```

Comment:
I don't think we should add this either. We should avoid changing the interfaces since it's a breaking change and this can be accomplished with out doing that. We can add a util method like `messageIDsEqual or MessageIDsEquals`.